### PR TITLE
CLN: avoid catching Exception in _choose_path

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by WillAyd** [comment]
If you are deep diving on this any chance you can add a docstring for func to help future readers?

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -646,20 +646,21 @@ def _choose_path(self, fast_path, slow_path, group):
...
-            pass
+            # Hard to know ex-ante what exceptions `fast_path` might raise
+            return path, res
+
+        # verify fast path does not change columns (and names), otherwise
+        # its results cannot be joined with those of the slow path
+        if not isinstance(res_fast, DataFrame):
+            return path, res
+
+        if not res_fast.columns.equals(group.columns):
```
Comment (WillAyd) [inline pandas/core/groupby/generic.py]: Can you add this condition to the check above?

**Review by WillAyd — CHANGES_REQUESTED** [review]
minor nit but otherwise lgtm - I think this certainly reads nicer

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/groupby/generic.py]: for coverage purposes Id rather leave them separate.  i.e. i want to be able to see that both cases are reached

**Comment by jreback** [comment]
lgtm. @WillAyd ?

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

**Pull request merged**